### ISSUE TITLE
CORS error

### ISSUE THREAD

**Issue by OmarMtya** [nathan-moore-97/NestJsCalculator#1]
Make sure that the URL that you're trying to hit has `http://` or `https://` at the begining.

**Comment by OmarMtya** [comment]
Sorry for leaving a comment here, Stackoverflow doesn't allow me to comment/give an answer in your post

**Comment by nathan-moore-97** [comment]
> Make sure that the URL that you're trying to hit has `http://` or `https://` at the begining.

That seemed to move the error. I added http:// to both the environment variable and the origin param in `main.ts` (as well as removing the trailing `/`)
 
![Error seems to be closer to what I want but is showing as a CORS problem still, which is confusing in its own right.](https://github.com/nathan-moore-97/NestJsCalculator/assets/27966747/0e640076-8399-4efb-b075-32f424905cb3)

I don't think I fully understand what the preflight OPTIONS are, what they do, and why they're used.

**Comment by OmarMtya** [comment]
If you try to hit that endpoint in Postman or any other HTTP Client, does it works?
Change `allow_origins` option to `*` and make sure that your endpoint is accessible

**Comment by nathan-moore-97** [comment]
> If you try to hit that endpoint in Postman or any other HTTP Client, does it works? Change `allow_origins` option to `*` and make sure that your endpoint is accessible

Yes, confirmed functional in Postman. For `allow_origins` do you mean this? 
 `app.enableCors({
    allowedHeaders: ['content-type'],
    origin: '*'
  });`

**Comment by OmarMtya** [comment]
Yes.

Ok, remove from `getListOfSupportedOpcodes()`  all extra headers. The method should look like this:
```
getListOfSupportedOpcodes() {
    return this.http.get(Enviornment.serverUrl + "/opcode");
  }
```

If you need to send a x-www-form-urlencoded, you can send a FormData object if needed. Like this:

```
let formData = new FormData();
formData.append('parameter', 'value');
return this.http.post(URL, formData);
```

I recommend you that if you need to use headers for every single call that you're making, look for an [Interceptor](https://angular.io/guide/http-interceptor-use-cases), if not, the simpler the better

**Comment by nathan-moore-97** [comment]
> If you need to send a x-www-form-urlencoded call, you can send a FormData object if needed.

That did the trick! I'll link this thread to the stack overflow thread soon. Can you tell me why my extra headers were causing a problem?

**Comment by OmarMtya** [comment]
> > If you need to send a x-www-form-urlencoded call, you can send a FormData object if needed.
> 
> That did the trick! I'll link this thread to the stack overflow thread soon. Can you tell me why my extra headers were causing a problem?

The headers `Access-Control-Allow-Origin` and `allow-origins` should be configured on the server side, not the client side.

Imagine that you are a bouncer at a bar and you have a policy that only allows entry to people from a specific address. Now, picture a scenario where a customer approaches the bar and asks you to let them in. They show you their identification (the handshake headers), which includes a note requesting entry (this note represents the client-supplied headers that were in that method). However, as the bouncer, you understand that it is ultimately the bar's decision to determine who gets to enter, not the customer's

**Comment by sandeepyadav605** [comment]
.`
/To access it from the front end locally you will have to change the main.ts enabled CORS settings.

From

app.enableCors({
  allowedHeaders: ['content-type'],
  origin: 'http://localhost:4200/'

});
To
app.enableCors(); //without any configuration option.`

//I have added the answer on the stack overflow, pls accept if it works for you.
